x/authz/client/testutil/grpc.go

```diff
@@ -176,40 +178,101 @@ func (s *IntegrationTestSuite) TestQueryGranterGrantsGRPC() {
 	}{
 		{
 			"invalid account address",
-			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/%s", val.APIAddress, "invalid address"),
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/granter/%s", val.APIAddress, "invalid address"),
 			true,
 			"decoding bech32 failed",
 			0,
 		},
 		{
 			"no authorizations found",
-			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/%s", val.APIAddress, grantee.String()),
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/granter/%s", val.APIAddress, grantee.String()),
 			false,
 			"",
 			0,
 		},
 		{
 			"valid query",
-			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/%s", val.APIAddress, val.Address.String()),
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/granter/%s", val.APIAddress, val.Address.String()),
 			false,
 			"",
 			7,
 		},
 		{
 			"valid query: expect seven grants",
-			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/%s", val.APIAddress, val.Address.String()),
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/granter/%s", val.APIAddress, val.Address.String()),
 			false,
 			"",
 			7,
 		},
 	}
 	for _, tc := range testCases {
 		s.Run(tc.name, func() {
-			resp, _ := rest.GetRequest(tc.url)
+			resp, err := rest.GetRequest(tc.url)
+			require.NoError(err)
+
 			if tc.expectErr {
 				require.Contains(string(resp), tc.errMsg)
 			} else {
-				var authorizations authz.QueryGrantsResponse
+				var authorizations authz.QueryGranterGrantsResponse
+				err := val.ClientCtx.Codec.UnmarshalJSON(resp, &authorizations)
+				require.NoError(err)
+				require.Len(authorizations.Grants, tc.numItems)
+			}
+
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestQueryGranteeGrantsGRPC() {
+	val := s.network.Validators[0]
+	grantee := s.grantee[1]
+	require := s.Require()
+
+	testCases := []struct {
+		name      string
+		url       string
+		expectErr bool
+		errMsg    string
+		numItems  int
+	}{
+		{
+			"invalid account address",
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/grantee/%s", val.APIAddress, "invalid address"),
+			true,
+			"decoding bech32 failed",
+			0,
+		},
+		{
+			"no authorizations found",
+			fmt.Sprintf("%s/cosmos/authz/v1beta1/grants/grantee/%s", val.APIAddress, grantee.String()),
+			false,
+			"",
+			0,
+		},
```

Comment:
Is this test case correct? I believe we did creat a grant for `grantee` in SetupSuite.

The error is that this grpc returns an array `[<something>]` with length 1, while lenght 0 is expected. The fact that `<something>` is so ugly is because of #10965.

Reply:
Right I switched them around and managed to get them to pass